Framework/Algorithms/src/ConvertUnits.cpp

```diff
@@ -734,8 +735,7 @@ API::MatrixWorkspace_sptr ConvertUnits::removeUnphysicalBins(
     MantidVec::difference_type bins = X0.cend() - start;
     MantidVec::difference_type first = start - X0.cbegin();
 
-    result =
-        WorkspaceFactory::Instance().create(workspace, numSpec, bins, bins - 1);
+    result = create<MatrixWorkspace>(*workspace, numSpec, BinEdges(bins));
```

Comment:
Note here (and in several other cases): `numSpec` is not required and can be omitted if the new size is equal to the parent size (that of `workspace` in this case).